core/src/HIP/Kokkos_HIP.cpp

```diff
@@ -146,6 +146,10 @@ void HIP::print_configuration(std::ostream& os, bool /*verbose*/) const {
 #else
   os << "no\n";
 #endif
+#ifdef KOKKOS_ENABLE_IMPL_HIP_UNIFIED_MEMORY
+  os << "  KOKKOS_ENABLE_IMPL_HIP_UNIFIED_MEMORY: ";
+  os << "yes\n";
+#endif
```

Comment:
```suggestion
 os << "  KOKKOS_ENABLE_IMPL_HIP_UNIFIED_MEMORY: ";
#ifdef KOKKOS_ENABLE_IMPL_HIP_UNIFIED_MEMORY
   os << "yes\n";
#else
   os << "no\n";
#endif
```

Reply:
Actually I like Christian's version better.
I would prefer if we would only print something when macros are defined and nothing otherwise, or at least that it looks closer to what the cmake-generated configuration file contains.  When we print more, we make the relevant part possibly less salient.
Maybe (not here) we could do something like only print undefined macros in verbose mode.
That said I don't want this discussion to delay progress on this work.